pinning.go

```diff
@@ -98,12 +98,23 @@ func (s *Server) doPinning(ctx context.Context, op *pinner.PinningOperation, cb
 	ctx, span := s.tracer.Start(ctx, "doPinning")
 	defer span.End()
 
+	connectedToAtLeastOne := false
 	for _, pi := range op.Peers {
 		if err := s.Node.Host.Connect(ctx, pi); err != nil {
 			log.Warnf("failed to connect to origin node for pinning operation: %s", err)
+		} else {
+			//	Check if it's trying to connect to itself since we only want to check if the
+			//	the connection is between the host and the external/other peers.
+			connectedToAtLeastOne = true
```

Comment:
If all addresses point to itself, I believe this will fail since they will all error out. Try this instead:
```suggestion
		if err := s.Node.Host.Connect(ctx, pi); err != nil && s.Node.Host.ID() != pi.ID {
			log.Warnf("failed to connect to origin node for pinning operation: %s", err)
		} else {
			//	Check if it's trying to connect to itself since we only want to check if the
			//	the connection is between the host and the external/other peers.
			connectedToAtLeastOne = true
```

Reply:
Please address this, so this PR can be merged